### PR TITLE
Fix Safari: only check selectionStart, selectionEnd on TEXTAREA or specific INPUT types

### PR DESCRIPTION
**Pull request by cpirich**
* @poorvasingal made an app that doesn't work on Safari: https://studio.code.org/projects/applab/MF6nTFkjSLUvvuGzuXztPQ (go to settings and try moving the slider)
* The `INPUT` element with type `range` (our slider control) is missing the `selectionStart` and `selectionEnd` properties that we expect we can safely read. It causes a type error exception. So now we follow the HTML spec and only request those properties from elements that are expected to provide them.

### CODE REVIEW

**Comment by islemaster** [comment]
Huh... it looks like this change would exclude our "textarea" design mode elements, which are actually contenteditable divs.  Any way we can include those?  Or will they just not work in Safari at all?

**Comment by cpirich** [comment]
@islemaster Good catch. I think they will work. I found this spec that includes TEXTAREA (will verify in Safari): https://dev.w3.org/html5/spec-preview/textFieldSelection.html

File: apps/src/applab/EventSandboxer.js [hunk 1]
```diff
@@ -130,8 +130,15 @@ EventSandboxer.prototype.sandboxEvent = function (event) {
       });
 
   // Pull selectionStart and selectionEnd from the target element, if available,
-  // and expose them on the sandboxed event object
-  if (event.target) {
+  // and expose them on the sandboxed event object. (Safari will throw if we
+  // we ask for these properties on elements that aren't these 5 INPUT types)
+  if (event.target && event.target.tagName === 'INPUT' &&
```
Comment (islemaster) [inline apps/src/applab/EventSandboxer.js]: So do we need to add `'TEXTAREA'` here?

**Comment by cpirich** [comment]
Note: these event properties don't work on Applab text areas (which are actually contenteditable DIVs). In the future, we might think about exposing the `selectionchange` event and making all of this work for contenteditable divs.

**Pull request merged**